entity/templates/src/main/java/package/domain/_Entity.java

```diff
@@ -224,6 +224,9 @@ public boolean equals(Object o) {
             return false;
         }
         <%= entityClass %> <%= entityInstance %> = (<%= entityClass %>) o;
+        if(<%= entityInstance %>.id == null || id == null) {
```

Comment:
The not obvious thing, is that if the ids are the same, then the entities are equals, but that is what Hibernate requires to function.